docs/reference/terraform.md

```diff
@@ -0,0 +1,299 @@
+
+## Workspaces
+
+The project uses [Terraform Workspaces](https://developer.hashicorp.com/terraform/cli/workspaces) to manage and deploy to multiple environments from the same codebase. Each workspace corresponds to a distinct environment and has its own state file, which prevents environments from interfering with each other.
+
+The workspaces are:
+
+- **`dev`**: Corresponds to the `dev` environment.
+- **`test`**: Corresponds to the `test` environment.
+- **`default`**: Corresponds to the `prod` environment.
+
+While each workspace is distinct with its own state file, all 3 state files are stored in the `prod` environment.
+
+The selection of the workspace is automated in the Azure DevOps pipeline. The `terraform/pipeline/workspace.py` script inspects the Git branch or tag that triggered the pipeline to determine which workspace to use. For example, a pull request to `main` will deploy to the `dev` environment, while a release tag will deploy to `test` or `prod`.
+
+For more details on each Azure environment, see the [Environments section in the Infrastructure reference](./infrastructure.md#environments).
+
+## Directory structure
+
+The `/terraform` directory is organized to separate concerns between the root configuration, reusable modules, and Azure DevOps pipeline logic.
+
+```text
+/terraform
+├── alerts.tf
+├── azure-pipelines.yml
+├── main.tf
+├── modules.tf
+├── security.tf
+├── variables.tf
+├── modules/
+│   ├── application/
+│   ├── database/
+│   ├── email/
+│   ├── key_vault/
+│   ├── monitoring/
+│   ├── network/
+│   └── storage/
+├── pipeline/
+│   ├── deploy.yml
+│   ├── tags.py
+│   └── workspace.py
+└── secrets/
+    ├── file.sh
+    ├── read.sh
+    └── value.sh
+```
+
+### Root files
+
+- `main.tf`: The entrypoint for the root module. It configures the Azure provider and backend state storage.
+- `variables.tf`: Defines input variables for the root module, such as subscription and object IDs.
+- `modules.tf`: The central wiring file that instantiates all the core modules from the `modules/` directory and connects their inputs and outputs.
+- `security.tf`: Contains centralized security configurations, such as Network Security Group (NSG) rules and Key Vault access policies that span across multiple modules.
+- `alerts.tf`: Defines monitoring alerts, such as the Application Insights error alert.
+
+### `modules/`
+
+This directory contains reusable, self-contained modules for different parts of the infrastructure. Each subdirectory is a separate Terraform module.
+
+### `pipeline/`
+
+Contains the definitions for the Azure DevOps pipeline that automates `terraform plan` and `terraform apply`.
+
+### `secrets/`
+
+Includes helper scripts for manually reading and writing secrets to Azure Key Vault.
+
+## Root modules
+
+The `.tf` files in the root of the `/terraform` directory define the core configuration and orchestrate the modules.
+
+### `main.tf`
+
+!!! abstract Source
```

Comment:
```suggestion
!!! abstract "Source"
```

Right now the block title is being rendered as "Abstract", but by making this change the block title will be rendered as "Source" which is probably what we want.
Similarly for all the other abstract blocks below.